adapters/handlers/grpc/v1/mapping.go

```diff
@@ -0,0 +1,257 @@
+		return &pb.Value{Kind: &pb.Value_ListValue{ListValue: &pb.ListValue{Values: []*pb.Value{}}}}, nil
+	}
+	val, ok := v.([]T)
+	if !ok {
+		return nil, protoimpl.X.NewError("invalid type: %T when serializing %v", v, dt.String())
+	}
+	list, err := NewPrimitiveList(val, dt)
+	if err != nil {
+		return nil, errors.Wrapf(err, "serializing array with type %v", dt)
+	}
+	return NewListValue(list), nil
+}
+
+// NewValue constructs a Value from a general-purpose Go interface.
+//
+//	╔════════════════════════╤════════════════════════════════════════════╗
+//	║ Go type                │ Conversion                                 ║
+//	╠════════════════════════╪════════════════════════════════════════════╣
+//	║ bool                   │ stored as BoolValue                        ║
+//	║ int, int32, int64      │ stored as NumberValue                      ║
+//	║ uint, uint32, uint64   │ stored as NumberValue                      ║
+//	║ float32, float64       │ stored as NumberValue                      ║
+//	║ string                 │ stored as StringValue; must be valid UTF-8 ║
+//	║ []byte                 │ stored as StringValue; base64-encoded      ║
+//	║ map[string]interface{} │ stored as StructValue                      ║
+//	║ []interface{}          │ stored as ListValue                        ║
+//	╚════════════════════════╧════════════════════════════════════════════╝
+//
+// When converting an int64 or uint64 to a NumberValue, numeric precision loss
+// is possible since they are stored as a float64.
+func NewPrimitiveValue(v interface{}, dt schema.DataType) (*pb.Value, error) {
+	innerDt, ok := schema.IsArrayType(dt)
+	if ok {
+		switch dt {
+		case schema.DataTypeBooleanArray:
+			return parseArray[bool](v, innerDt)
+		case schema.DataTypeDateArray:
+			return parseArray[string](v, innerDt)
+		case schema.DataTypeNumberArray:
+			return parseArray[float64](v, innerDt)
+		case schema.DataTypeIntArray:
+			return parseArray[float64](v, innerDt)
+		case schema.DataTypeStringArray:
+			return parseArray[string](v, innerDt)
+		case schema.DataTypeTextArray:
+			return parseArray[string](v, innerDt)
+		case schema.DataTypeUUIDArray:
+			return parseArray[string](v, innerDt)
+		default:
+			return nil, protoimpl.X.NewError("invalid type: %T", v)
+		}
+	} else {
+		switch dt {
+		case schema.DataTypeBoolean:
+			val, ok := v.(bool)
+			if !ok {
+				return nil, protoimpl.X.NewError("invalid type: %T expected bool when serializing bool", v)
+			}
+			return NewBoolValue(val), nil
+		case schema.DataTypeDate:
+			val, ok := v.(string)
+			if !ok {
+				return nil, protoimpl.X.NewError("invalid type: %T expected string when serializing date", v)
+			}
+			return NewDateValue(val), nil
+		case schema.DataTypeNumber:
+			val, ok := v.(float64)
+			if !ok {
+				return nil, protoimpl.X.NewError("invalid type: %T expected float64 when serializing number", v)
+			}
+			return NewNumberValue(val), nil
+		case schema.DataTypeInt:
```

Comment:
more types are missing here - can you add tests for each type? Could be a parameterized test